Blog/include/functions.php

```diff
@@ -44,16 +44,38 @@ function getComments($blogPostID){ //colon is a security prevention
 		WHERE blogPostID = :blogPostID
 	",
 	array('blogPostID' => $blogPostID)
-);
+)->fetchAll();
 
-	return $result->fetchAll();
+	return $result;
 
 }
 
 function printHeader(){
 		echo "<html>";
 };
 
+@$userID = $_REQUEST['userID'];
+@$Comment = $_REQUEST['Comment'];
+
+//var_dump($_REQUEST);
 
+function insertBlogComment($blogPostID, $userID, $commentTimeStamp, $Comment){
+	if(isset($_REQUEST['submitComment'])){
```

Comment:
You shouldn't need to check this inside the function. Presumably you already checked that on whatever web page calls this, and you wouldn't be calling this function in the first place if the form hasn't been submitted.

The goal with functions should be to abstract away an understanding about what's going on elsewhere in the code. Functions take parameters, maybe run a query, and maybe return an output. They shouldn't need to rely on anything outside of the function (like $_REQUEST) in most cases.